app/archive/amazing-race-35/league-standing/page.tsx

```diff
@@ -1,18 +1,20 @@
 import { WIKI_API_URL } from '../../../leagueConfiguration/AmazingRace_35'
 import { CONTESTANT_LEAGUE_DATA } from '../../../leagueData/AmazingRace_35'
 import { getWikipediaContestantDataFetcher } from '../../../utils/wikiFetch'
-import generateContestantRoundScoreComponent from '../../../generators/contestantRoundScoreGenerator'
+import { generateContestantRoundScores } from '../../../generators/contestantRoundScoreGenerator'
+import LeagueStandingTable from '../../../components/leagueStandingTable/leagueStandingTable';
 
 export default async function LeagueStanding() {
 
     const dataFetcher = getWikipediaContestantDataFetcher(WIKI_API_URL, "Cast")
-    const contestantScoresComponent = await generateContestantRoundScoreComponent(dataFetcher, CONTESTANT_LEAGUE_DATA)
-
+    const contestantScoresData = await generateContestantRoundScores(dataFetcher, CONTESTANT_LEAGUE_DATA);
+    const contestantScores = contestantScoresData.rounds[contestantScoresData.rounds.length-1].contestantRoundData;
```

Comment:
fft: Is there not a better way of doing this in javascript in 2024?
Like what about with `.slice()`
```suggestion
    const contestantScores = contestantScoresData.rounds.slice(-1)[0].contestantRoundData;
```
or maybe even `.at`
```suggestion
    const contestantScores = contestantScoresData.rounds.at(-1).contestantRoundData;
```
Not exactly sure if either of those have drawbacks, besides just being fresh syntax.

Reply:
Also just putting this out there, but this might be a good time to start talking about back-end vs front-end and how to do better templating. I was trying to put as much of this logic in that generator (even including the components) so that the pages are as simple between leagues making the copy and paste job as minimal as possible. Obviously part of that problem is copy and paste, so that obviously has it's own solution. But the other part is if we only want to display the most recent results perhaps it is worth considering if the generator could take that as an argument or an option. WDYT?

Reply:
I agree. There's a bit of overall cleanup we should do. I think after user accounts, we should take the time to refactor and clean up

Reply:
Ok, but what about my first suggestions?